app/models/service.py

```diff
@@ -425,6 +427,14 @@ def get_days_of_retention(self, notification_type):
             "days_of_retention", current_app.config["ACTIVITY_STATS_LIMIT_DAYS"]
         )
 
+    def get_consistent_data_retention_period(self) -> Optional[int]:
```

Comment:
Save going to Redis every time this is called:
```suggestion
    @cached_property
    def consistent_data_retention_period(self) -> Optional[int]:
```

Reply:
Doesn't hurt to do this, but it uses `get_days_of_retention`, which accesses `self.data_retention`, which is already a cached property, so we already only hit the API/redis once per call.